app/web/src/workers/types/dbinterface.ts

```diff
@@ -285,13 +287,75 @@ export interface ComponentQualificationTotals {
   running: number;
 }
 
-export interface Component {
+export interface EddaComponent {
   id: ComponentId;
   name: string;
   color?: string;
   schemaName: string;
   schemaId: SchemaId;
-  schemaVariantId: SchemaVariantId;
+  schemaVariantId: Reference;
+  schemaVariantName: string;
+  schemaVariantDescription?: string;
+  schemaVariantDocLink?: string;
+  schemaCategory: string;
+  hasResource: boolean;
+  qualificationTotals: ComponentQualificationTotals;
+  inputCount: number;
+  // this will only be filled in when it is computed
+  outputCount: number;
```

Comment:
I'll drop this in a follow-up PR. We'll remove the logic from edda / dal-materialized views to compute this make MV generation faster too.